File: fullstack/MusicAPI/MusicAPI.Models/Song.cs
using Microsoft.EntityFrameworkCore;

namespace MusicAPI.Models
{
    public class Song
    {
        public int Id { get; set; }
        public string Title { get; set; } = "";
        public string Url { get; set; } = "";
        public List<Artist> Artists { get; set; } = new List<Artist>();
        public int RatingCount { get; set; }
        public int TotalRating { get; set; }
        public double AverageRating { get; set;}
        public Genre Genre { get; set; }
        public int Tempo { get; set; }
    }
}

File: fullstack/MusicUI/src/app/Models/song.ts
import { Artist } from "./artist";
import { Genre } from "./genre";

export interface Song {
    id: number;
    title: string;
    url: string;
    artists: Artist;
    ratingCount: number;
    totalRating: number;
    averageRating: number;
    genre: Genre;
    tempo: number
}

/*
{
    "id": 0,
    "title": "string",
    "url": "string",
    "artists": [
      {
        "id": 0,
        "name": "string",
        "songs": [
          "string"
        ]
      }
    ],
    "ratingCount": 0,
    "totalRating": 0,
    "averageRating": 0,
    "genre": {
      "id": 0,
      "name": "string",
      "song": [
        "string"
      ]
    },
    "tempo": 0
  }
*/